components/script/dom/webglshader.rs

```diff
@@ -284,44 +169,36 @@ impl WebGLShader {
-                // Work around that Intel drivers on Mac OSX handle for-loop incorrectly.
-                mozangle::shaders::ffi::SH_ADD_AND_TRUE_TO_LOOP_CONDITION
-            } else {
-                // We want to do this everywhere, but to do this on Mac, we need
-                // to do it only on Mac OSX > 10.6 as this causes the shader
-                // compiler in 10.6 to crash
-                mozangle::shaders::ffi::SH_CLAMP_INDIRECT_ARRAY_BOUNDS
-            };
-
-        // Replicating
-        // https://github.com/servo/mozangle/blob/706a9baaf8026c1a3cb6c67ba63aa5f4734264d0/src/shaders/mod.rs#L226
-        let options = options |
-            mozangle::shaders::ffi::SH_VALIDATE |
-            mozangle::shaders::ffi::SH_OBJECT_CODE |
-            mozangle::shaders::ffi::SH_VARIABLES | // For uniform_name_map()
-            mozangle::shaders::ffi::SH_EMULATE_ABS_INT_FUNCTION | // To workaround drivers
-            mozangle::shaders::ffi::SH_EMULATE_ISNAN_FLOAT_FUNCTION | // To workaround drivers
-            mozangle::shaders::ffi::SH_EMULATE_ATAN2_FLOAT_FUNCTION | // To workaround drivers
-            mozangle::shaders::ffi::SH_CLAMP_INDIRECT_ARRAY_BOUNDS |
-            mozangle::shaders::ffi::SH_INIT_GL_POSITION |
-            mozangle::shaders::ffi::SH_ENFORCE_PACKING_RESTRICTIONS |
-            mozangle::shaders::ffi::SH_LIMIT_EXPRESSION_COMPLEXITY |
-            mozangle::shaders::ffi::SH_LIMIT_CALL_STACK_DEPTH;
+        // https://searchfox.org/mozilla-esr115/rev/f1fb0868dc63b89ccf9eea157960d1ec27fb55a2/dom/canvas/WebGLShaderValidator.cpp#29
+        let mut options = CompileOptions::mozangle();
+        options.set_variables(1);
+        options.set_enforcePackingRestrictions(1);
+        options.set_objectCode(1);
+        options.set_initGLPosition(1);
+        options.set_initializeUninitializedLocals(1);
+        options.set_initOutputVariables(1);
+
+        options.set_limitExpressionComplexity(1);
+        options.set_limitCallStackDepth(1);
+
+        if cfg!(target_os = "macos") {
+            options.set_removeInvariantAndCentroidForESSL3(1);
+
+            // Work around https://bugs.webkit.org/show_bug.cgi?id=124684,
+            // https://chromium.googlesource.com/angle/angle/+/5e70cf9d0b1bb
+            options.set_unfoldShortCircuit(1);
+            // Work around that Mac drivers handle struct scopes incorrectly.
+            options.set_regenerateStructNames(1);
+            // TODO: Only apply this workaround to Intel hardware
+            // Work around that Intel drivers on Mac OSX handle for-loop incorrectly.
+            options.set_addAndTrueToLoopCondition(1);
+            options.set_rewriteTexelFetchOffsetToTexelFetch(1);
+        } else {
+            // We want to do this everywhere, but to do this on Mac, we need
+            // to do it only on Mac OSX > 10.6 as this causes the shader
+            // compiler in 10.6 to crash
+            options.set_clampIndirectArrayBounds(1);
+        }
```

Comment:
I'm pretty sure we only support MacOS 10.10 and up, so this if statement can likely be removed in a followup.